plugins/inputs/f5_load_balancer/f5_load_balancer.go

```diff
@@ -0,0 +1,343 @@
+package f5_load_balancer
+
+import (
+	"bytes"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+	"strings"
+	"sync"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"github.com/tidwall/gjson"
+)
+
+type F5LoadBalancer struct {
+	Username   string   `toml:"username"`
+	Password   string   `toml:"password"`
+	URL        string   `toml:"url"`
+	Collectors []string `toml:"collectors"`
+	Token      string
+	Log        telegraf.Logger
+}
+
+const sampleConfig = `
+  ## F5 Load Balancer Username
+  username = "" # required
+  ## F5 Load Balancer Password
+  password = "" # required
+  ## F5 Load Balancer User Interface Endpoint
+  url = "https://f5.example.com/" # required
+  ## Metrics to collect from the F5
+  collectors = ["node","virtual","pool","net_interface"]
+`
+
+// Description will appear directly above the plugin definition in the config file
+func (f5 *F5LoadBalancer) Description() string {
+	return `Gathers metrics from an F5 Load Balancer's API`
+}
+
+// SampleConfig will populate the sample configuration portion of the plugin's configuration
+func (f5 *F5LoadBalancer) SampleConfig() string {
+	return sampleConfig
+}
+
+func (f5 *F5LoadBalancer) Init() error {
+	if f5.Username == "" {
+		return fmt.Errorf("Username cannot be empty")
+	}
+	if f5.Password == "" {
+		return fmt.Errorf("Password cannot be empty")
+	}
+	if f5.URL == "" {
+		return fmt.Errorf("URL cannot be empty")
+	}
+	if f5.Collectors == nil {
+		f5.Collectors = []string{"node", "virtual", "pool", "net_interface"}
+	}
+	f5.Token = ""
+
+	return nil
+}
+
+func (f5 *F5LoadBalancer) Gather(acc telegraf.Accumulator) error {
+	// Due to self signed certificates in many orgs, we don't verify the cert
+	http.DefaultTransport.(*http.Transport).TLSClientConfig = &tls.Config{InsecureSkipVerify: true}
```

Comment:
Telegraf provides a helpful utility to configure tls, you can use the package: `"github.com/influxdata/telegraf/plugins/common/tls"` and then add `tls.ClientConfig` to your struct and then the user can configure the tls certificate.

You can find an example of using the tls config utility in this Jenkins plugin: https://github.com/influxdata/telegraf/blob/31ec509844959258fb718601225e18f8065a753a/plugins/inputs/jenkins/jenkins.go

On another note, you should also default to false here to verify the certificate. Here is a link to the configuration you'd want: https://github.com/influxdata/telegraf/blob/31ec509844959258fb718601225e18f8065a753a/plugins/inputs/jenkins/jenkins.go#L59-L64